### ISSUE TITLE
Menu icon with no menu

### ISSUE THREAD

**Issue by nwoolls** [rdsubhas/docker-menu#2]
Node.js: `4.2.1`
OS X: `El Capitan`

I cloned the repo and ran `npm install` followed by `grunt` which built the .app file. I ran the .app file and get the icon in my menu bar, but clicking it reveals no menu.

**Comment by EtreRe** [comment]
Same version and same issue for me.

**Comment by mauilion** [comment]
here here

A little more info:

When I run the command manually I see:

dcooley@lynx ~
$ /Applications/DockerMenu-0.2.0-Mac.app/Contents/Frameworks/Electron\ Framework.framework/Resources/crashpad_handler --handshake-fd=20 --database=/tmp/docker-menu\ Crashes --url=http://54.249.141.255:1127/post
[1023/151138:ERROR:file_io.cc(26)] read: Bad file descriptor
[1023/151138:FATAL:file_io.cc(53)] Check failed: LoggingReadFile(file, buffer, size).
Trace/BPT trap: 5

**Comment by rdsubhas** [comment]
Thanks @nwoolls @EtreRe @mauilion I'm looking into this. Btw can I assume all of you are on `El Capitan`?

**Comment by EtreRe** [comment]
Yes, I'm on El Capitain.

**Comment by rdsubhas** [comment]
(wrong tag on above commit, pls ignore)

**Comment by rdsubhas** [comment]
@nwoolls @EtreRe @mauilion I've pushed a few fixes for the above problem. Can you try running `npm start` and see if it works fine?

**Comment by nwoolls** [comment]
I get a menu now, but it has `Error fetching data!` in it.

**Comment by rdsubhas** [comment]
I'll try out other fixes tomorrow. I'm guessing this has something to do with el Capitan. I'm still on Yosemite. thanks for helping test this...

**Comment by rdsubhas** [comment]
@nwoolls @EtreRe @mauilion  if I can ask one help :) It would be very useful to have a full stack trace as this issue is driving me crazy. Can you please run `npm install && npm start` and print the full error in the console?

I've pushed various fixes - updating Electron, disabling the crash reporter, disabling TLS verification and more. A full stack trace would really help.

EDIT: Missing `docker-machine` might also be an issue. I've added a dialog box to locate docker-machine when its not found.

**Comment by nwoolls** [comment]
With the new version:
- If I run via `npm start` there is no error, the app starts, the menu icon turns blue, and I can see a list of containers.
- If I run `grunt` and then double-click the resulting .app file, the menu icon never turns blue and I do not see a list of containers.

So maybe the difference is in running the .app versus running from the CLI?

**Comment by rdsubhas** [comment]
Nice catch! Could be the packaging. I'll cross check the grunt tasks :+1:

**Comment by jsrraman** [comment]
Seems to be an issue with El Capitan, something has changed when Finder/Spotlight launches an app (through `launchd` maybe?)

When running `npm start`, the output of `docker-machine` from `lib/docker.js` is:

```
NAME   ACTIVE   DRIVER       STATE     URL                         SWARM
dev             virtualbox   Running   tcp://192.168.99.100:2376   
```

The same `npm start` output when launched through Finder:

```
NAME   ACTIVE   DRIVER       STATE   URL   SWARM
dev             virtualbox   Error         
```

**Comment by jsrraman** [comment]
Launching the grunt-built Docker Menu app via terminal works fine! Just opening it via Finder alone doesn't work!!

**Comment by rdsubhas** [comment]
**Phew, finally found the root cause of this issue**

When an app is launched via Finder, OSX 10.11 prunes the PATH to just `/usr/bin:/bin:/usr/sbin:/sbin`. Everything else like `/usr/local/bin` are removed. This raises all sorts of problems up until `VBoxManage` is not found.

We have to manually fix the PATH when running in OSX 10.11 :disappointed: Expect a fix soon. Thanks to @jsrraman!

**Comment by rdsubhas** [comment]
Closed too early, waiting for some more validation before I close this issue.

**Comment by rdsubhas** [comment]
Verified and fixed in `v0.3.0`
